### ISSUE TITLE
Simplify login procedure

### ISSUE THREAD

**Issue by AntonioFasano** [keepassxreboot/keepassxc-browser#1502]
## Current Behaviour

If my understanding is correct (correct me, please, if I am wrong), to login to the site "foo.com" (`foo` being a unique Keepass entry), the steps are:

1. Give focus to KeepassXC Window (shortcut depends on your system),
2. `Ctrl-F`,
3. `foo`,
4. `Ctrl-Shift-U`,
5. `Alt-Shift-U` `Enter`.

In  commercial products, such as LastPass, you can simply type:

`Alt-W` `foo` `Enter`

and similarly in Bitwarden.

## Desired Behaviour

LP approach seems very efficient here.

More generally, the extension should be able to access the whole database from a new browser tab, so we could make the search from there.

**Comment by droidmonkey** [comment]
The workflow should be:

1. Go to desired site
2. Click keepassxc icon or press keyboard shortcut
3. Credentials are entered, login

This requires you to add the site's url to the entry associated with that site. You can have more than one entry associated to a site, a pick list will be shown. At no time do you need to interact with keepassxc or use a search if that linkage is setup.

**Comment by AntonioFasano** [comment]
@droidmonkey, sure, but the point is: how do I go to the desired site?

There are three options:

1) use a search engine to find the address,
2) use browser bookmarks,
3) use KeepassXC to find the site.



To me  1) is definitely  time-consuming.   
2)  means to maintain (create, organise, backup, sync) both browser bookmarks and Keepass items (where the latter are partly a duplicate of the former). This is super complex, particularly for a huge list of sites organised in folders.   
3) seems the only viable option to me, and it implies 1-3 in my original  question.

**Comment by droidmonkey** [comment]
That is your choice really. By design we do not allowing searching the database from the extension. We have chosen a much more isolated approach then bitwarden and others to greatly improve security.

**Comment by AntonioFasano** [comment]
I see. Is the app calling  the extension and asking to login to a certain site still unsecure?

**Comment by varjolintu** [comment]
@AntonioFasano If you have URL set to your entry, you can still normally open the URL from KeePassXC and let do the filling using the extension icon, or keyboard shortcut.

**Comment by jekuno** [comment]
I think, the suggestion of @AntonioFasano would greatly improve the usability of keepassxc. I switched from 1Password to keepassxc. 1Password has the same feature like Bitwarden, LastPass and others:

You browse the internet and want to open a certain page. You don't have to leave your browser to do so. Just press the "fill username+password" shortcut directly from your browser. You then see a small search box, type "foo" and press enter to open the first search result, such as the login page of foo.com.

1Password then even fills in the credentials for foo.com automatically as soon as the page is loaded. This is very convenient, but as a first step, it would be sufficient if keepassxc would provide a search box to open foo.com from the browser. I could then manually press my "fill username+password" shortcut a second time to login.

Advantages of opening keepassxc URLs directly from the browser:
1. Stay in your browser - instead of manually switching to keepassxc => Save time
2. Automatically open URL in your currently active browser = the correct browser (I use keepassxc in both Firefox and Chrome)
3. Motivate users to switch from LastPass, 1Password, Bitwarden and others to keepassxc by providing a replacement for their daily workflow (in 1Password I opened EVERY page with a login using that workflow; it's a big time saver and improves usability)

**Comment by varjolintu** [comment]
@jekuno This can be already achieved with Global Auto-Type.

**Comment by jekuno** [comment]
@varjolintu I tried that. It allows me to search the saved keepassxc entries but it doesn't open the URL of the selected entry in the browser. Instead it simply pastes my username and password into the current browser page.

**Comment by varjolintu** [comment]
@jekuno This is because we don't allow access to all credentials by design. There's one [pull request](https://github.com/keepassxreboot/keepassxc/pull/7292) that adds a feature for this kind of possibility.

**Comment by jekuno** [comment]
@varjolintu Perfect. The mentioned [pull request](https://github.com/keepassxreboot/keepassxc/pull/7292) seems to form the basis for the function requested here. Great that this is already in the works.

Out of curiosity: What's the reason that you don't allow to access all credentials by default? How could that feature be exploited if you control both the browser addon and keypassxc itself and if the user can only invoke that feature manually by pressing a predefined shortcut?

**Comment by varjolintu** [comment]
> Out of curiosity: What's the reason that you don't allow to access all credentials by default? How could that feature be exploited if you control both the browser addon and keypassxc itself and if the user can only invoke that feature manually by pressing a predefined shortcut?

Denying access to all credentials by default decreases the possible attack surface. Web pages have only access to credentials that matches the site itself. Although I'm not aware of actual attacks that could be done from a fraudulent page etc. (the access to extension data shouldn't be possible even from scripts), but if an option for acccessing all credentials is enabled, in theory malware applications could try accessing the database outside the browser and/or extension.

So in short the answer to your question is: even if you control the extension and KeePassXC, the API that is used for retrieving the credentials can be accessed by any local application.

This is why we also encourage using the Access Confirm option, which is already enabled by default. Meaning even if some malware could get your connection keys etc. a manual confirmation for accessing the password is still required.

**Comment by jekuno** [comment]
Thanks for the explanation. My hope was, that any application, that wants to access that API, needs to be authorized by the user from within keypassxc once initially, so that it would be technically ensured, that e.g. only the confirmed browser extensions get access to that API.

If that's not the case, I could image that the folowing workflow would be better:

1. User surfs in browser and wants to open a certain website
2. User presses global shortcut (similar to Global Auto-Type)
3. Tiny bookmark search window of keepassxc (and not of the browser add-on) opens
4. User selects matching entry
5. keepassxc sends selected entry (including data such as URL, username and password) to the browser extension of the currently active browser window
6. Browser opens the according URL and signs in
